src/tools/MappingTools.ts

```diff
@@ -41,32 +42,70 @@ export function mapFile(sourceFile: ts.SourceFile) {
           break;
         case ts.SyntaxKind.InterfaceDeclaration:
           file.addInterface(
-            mapInterface(<ts.InterfaceDeclaration>child, sourceFile),
+            mapInterface(<ts.InterfaceDeclaration>child, sourceFile)
           );
           break;
         case ts.SyntaxKind.VariableStatement:
           file.addVariable(
-            mapVariableStatement(<ts.VariableStatement>child, sourceFile),
+            mapVariableStatement(<ts.VariableStatement>child, sourceFile)
           );
           break;
         case ts.SyntaxKind.FunctionDeclaration:
           file.addFunction(
-            mapFunction(<ts.FunctionDeclaration>child, sourceFile),
+            mapFunction(<ts.FunctionDeclaration>child, sourceFile)
           );
           break;
         case ts.SyntaxKind.EnumDeclaration:
           file.addEnum(mapEnums(<ts.EnumDeclaration>child));
           break;
         case ts.SyntaxKind.ExpressionStatement:
           file.addExpression(
-            mapExpressions(<ts.ExpressionStatement>child, sourceFile),
+            mapExpressions(<ts.ExpressionStatement>child, sourceFile)
           );
           break;
+        case ts.SyntaxKind.ExportKeyword:
+          let lineText: string[] = sourceFile.getText().split('\n');
+          file.addExport(
+            mapExportKeyword(file, lineText[counter])
+          );
+          counter++;
```

Comment:
What is the purpose of this counter? It seems to me that it is only used once? Is this function called in some kind of loop?

Reply:
Yes. It is called in a foreach-Loop and is incremented in this loop.